### PR TITLE
Save UserHistoryLog, update state of Procedure in db

### PR DESCRIPTION
**Pull request by EvgSem**

### CODE REVIEW

File: FaceCareRoutine/AppDelegate.swift [hunk 1]
```diff
@@ -10,10 +10,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         //get path of dataBase
         let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
         print(paths[0])
-        
-        _ = DataBaseManagement.shared
-        let dataInit = DataBaseInitializer(dbInput: DataBaseManagement.shared.db!)
-        dataInit.initiateDataBase()
+        if DB {
```
Comment (sbondini) [inline FaceCareRoutine/AppDelegate.swift]: it is better to give more meaningful names to the constants

File: FaceCareRoutine/AppDelegate.swift [hunk 2]
```diff
@@ -10,10 +10,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+            _ = DataBaseManagement.shared
```
Comment (sbondini) [inline FaceCareRoutine/AppDelegate.swift]: why reading variable and not storing it in a variable?

File: FaceCareRoutine/CalendarTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    func setProcedure(procedure: Procedure){
+        self.procedure = procedure
+    }
+    
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+    }
+    
+    @IBAction func switchChanged(_ sender: Any) {
+        delegate?.didSwitchChanged(procedure: procedure, state: isDone.isOn)
```
Comment (sbondini) [inline FaceCareRoutine/CalendarTableViewCell.swift]: what if procedure is nil, when this method is called?

File: FaceCareRoutine/CalendarViewController.swift [hunk 4]
```diff
@@ -119,6 +114,13 @@ class CalendarViewController: UIViewController, UITableViewDelegate, UITableView
 
 }
 
+//MARK: Table save state isDONE of Procedures
+extension CalendarViewController: calendarProcedureProtocol{
```
Comment (sbondini) [inline FaceCareRoutine/CalendarViewController.swift]: extract extensions to separate file (ie CalendarViewControllerExtensions)

File: FaceCareRoutine/Procedure.swift [hunk 5]
```diff
@@ -20,6 +21,7 @@ class Procedure{
 
     init(){
         self.order = 0
+        self.isNight = false
```
Comment (sbondini) [inline FaceCareRoutine/Procedure.swift]: Better create enum with two cases: Day, Night

File: FaceCareRoutine/ComplexViewController.swift [hunk 6]
```diff
@@ -121,7 +114,9 @@ class ComplexViewController: UIViewController, UITableViewDelegate, UITableViewD
                 ProcedureDetailViewController.procedure = selectedProcedure
             case "showCalendar":
                 //save all procedures
-                complexFactory?.complex.saveProcedures()
+                if DB {
```
Comment (sbondini) [inline FaceCareRoutine/ComplexViewController.swift]: would it be more convenient to move this check inside "saveProcedures" method?

File: FaceCareRoutine/ProcedureLog.swift [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import Foundation
+class ProcedureLog {
+    var date: Date
+    var procedureOrderPriority : Int
+    var isNight: Bool
+    var isDone: Bool
```
Comment (sbondini) [inline FaceCareRoutine/ProcedureLog.swift]: Any chance there might be more that two states for the procedure? (ie "scheduled", "in progress" etc)

File: FaceCareRoutine/ProcedureViewController.swift [hunk 8]
```diff
@@ -157,8 +139,7 @@ class ProcedureViewController: UIViewController, UITextFieldDelegate,  UIPickerV
             os_log("The save button was not pressed, cancelling", log: OSLog.default, type: .debug)
             return
         }
-       // procedure = Procedure()
-        procedure = Procedure(order: 1, careType: CareType.mandatory  , howOften: getHowOften(), procedureType:  ProcedureType(rawValue: getName())!, duration: 2, instruction: getInstructionName())
+        procedure = Procedure(order: 1, isNight: false, careType: CareType.mandatory  , howOften: getHowOften(), procedureType:  ProcedureType(rawValue: getName())!, duration: 2, instruction: getInstructionName())
```
Comment (sbondini) [inline FaceCareRoutine/ProcedureViewController.swift]: The problem with creating objects that has such big amount of parameters is risk, that at some point you change constructor signature. You will need to go through all the places where procedure is created and fix it.

Comment (sbondini) [reply FaceCareRoutine/ProcedureViewController.swift]: Try moving creation to separate class

File: FaceCareRoutine/UserSkinViewController.swift [hunk 9]
```diff
@@ -182,13 +182,11 @@ class UserSkinViewController: UIViewController, UIPickerViewDelegate, UIPickerVi
     
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         if (segue.identifier == "showComplex") {
-            // pass data to next view
             setUserSkin()
-            self.userSkin?.save()
-            
+            if DB {
+                self.userSkin?.save()
```
Comment (sbondini) [inline FaceCareRoutine/UserSkinViewController.swift]: Could "DB" be moved to save method?

File: FaceCareRoutine/ProcedureLog.swift (hunk 7 above)
Comment (EvgSem) [reply FaceCareRoutine/ProcedureLog.swift]: No

**Pull request merged**